toolkit/tools/scheduler/schedulerutils/graphbuildstate.go

```diff
@@ -143,10 +176,22 @@ func (g *GraphBuildState) RecordBuildResult(res *BuildResult, allowToolchainRebu
 		g.failures = append(g.failures, res)
 	}
 
+	// 'NodeFreshnessRebuildRequired' is a special value that indicates that the node was rebuilt due to  missing files
+	// (user requested rebuilds are already at the max freshness). In this case, we want to reset the freshness to the
+	// max, so that subsequent dependant nodes will be rebuilt. Also ensure that the freshness is not greater than the max.
+	freshness := res.ActualFreshness
+	if freshness < 0 || freshness > g.GetMaxFreshness() {
```

Comment:
These unusual values of freshness look like something, that should never happen and is a bug in our code. For such cases I think we should just error-out, since we no longer can say we're in a valid state.

Reply:
Done